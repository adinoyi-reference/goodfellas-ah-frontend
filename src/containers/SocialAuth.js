import React, { Component } from 'react';
import { connect } from 'react-redux';
import queryString from 'query-string';
import { socialSignin, showSignError } from '../actions/authActions';
import LargeLoader from '../components/shared/LargeLoader';

const socialAuthError = 'You can\'t login through this platform';

class SocialAuth extends Component {
  componentWillMount() {
    const {
      socialSignin: socialSigninUser,
      showSignError: showSocialSigninError,
      history,
    } = this.props;
    const { token, userId, error } = queryString.parse(history.location.search);
    if (error === 'true') {
      showSocialSigninError(socialAuthError, () => history.push('/auth/signin'));
    } else {
      socialSigninUser({ token, userId }, (response) => {
        if (response) {
          history.push('/user/profile');
        } else {
          history.push('/auth/signin');
        }
      });
    }
  }

  render() {
    return (
      <div>
        <LargeLoader />
      </div>
    );
  }
}

export default connect(null, { socialSignin, showSignError })(SocialAuth);
