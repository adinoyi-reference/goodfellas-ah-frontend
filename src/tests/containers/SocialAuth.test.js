import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Root from '../../root';
import SocialAuth from '../../containers/SocialAuth';

let wrapped;
const history = {
  location: {
    search: {
      token: 'token',
      userId: 'userId',
    },
  },
};

beforeEach(() => {
  wrapped = mount(
    <Root>
      <SocialAuth history={history} />
    </Root>,
  );
});

afterEach(() => wrapped.unmount());

describe('SocialAuth UI', () => {
  describe('render features', () => {
    test('container should render as expected', () => {
      const tree = toJson(wrapped);
      expect(tree).toMatchSnapshot();
    });

    test('should conmtain a loader', () => {
      expect(wrapped.exists('.large-spinner')).toEqual(true);
    });
  });
});
