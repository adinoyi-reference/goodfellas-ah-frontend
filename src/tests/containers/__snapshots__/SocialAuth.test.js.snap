// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialAuth UI render features container should render as expected 1`] = `
<div>
  <LargeLoader />
</div>
`;
